File: exercism/scrabble_score.clj
(defn score-letter [c]
  (case c
    (\d \g) 2
    (\b \c \m \p) 3
    (\f \h \v \w \y) 4
    (\k) 5
    (\j \x) 8
    (\q \z) 10
    1))

(defn score-word [word]
  (reduce 
    (fn [acc c]
      (+ acc (score-letter c)))
    0
    (seq (clojure.string/lower-case word))))

File: exercism/scrabble_score.py
def score(word):
    letter_scores = {
        "aeioulnrst": 1,
        "dg": 2,
        "bcmp": 3,
        "fhvwy": 4,
        "k": 5,
        "jx": 8,
        "qz": 10,
    }

    def letter_score(letter):
        for letters in letter_scores:
            if letter in letters:
                return letter_scores[letters]
        return 0
    
    return sum(letter_score(letter) for letter in word.lower())
